server/src/test/java/org/elasticsearch/index/codec/vectors/cluster/HierarchicalKMeansTests.java

```diff
@@ -127,7 +127,7 @@ public void testFewDifferentValues() throws IOException {
             assertEquals(nVectors, soarAssignments.length);
             // verify no duplicates exist
             for (int i = 0; i < assignments.length; i++) {
-                assertTrue(soarAssignments[i] >= 0 && soarAssignments[i] < centroids.length);
+                assertTrue(soarAssignments[i] >= -1 && soarAssignments[i] < centroids.length);
```

Comment:
minor: I wonder if it's useful to move that special value into a constant, so that we can reuse the same "too close" value both in code and tests?